### PR TITLE
Chores/static deploy

### PR DESCRIPTION
**Pull request by will-wow**
# Problem

Static files weren't deploying

# Solution

Add `whitenoise`. 
Also fix up the dockerfile (it wasn't running, not sure when that broke)

## Change summary

- Switch to a docker image that's easier to use with python
- Add whitenoise
- Run collectstatic in the dockerfile (I coudn't get it working in an EB `container_command`)

## Screenshot (Optional)

Screenshot from 2019-11-12 01-51-57

### CODE REVIEW

File: confero/settings.py [hunk 1]
```diff
@@ -129,5 +130,6 @@
 # Static files (CSS, JavaScript, Images)
 # https://docs.djangoproject.com/en/2.1/howto/static-files/
 
-STATIC_ROOT = os.path.join(BASE_DIR, "..", "www", "static")
+STATIC_ROOT = os.path.join(BASE_DIR, 'staticfiles')
 STATIC_URL = '/static/'
+STATICFILES_STORAGE = 'whitenoise.storage.CompressedManifestStaticFilesStorage'
```
Comment (will-wow) [inline confero/settings.py]: Dang that was easy, good call on whitenoise!

File: confero/templates/confero/base.html [hunk 2]
```diff
@@ -21,8 +21,6 @@
       crossorigin="anonymous"
     />
 
-    <link rel="stylesheet" href="{% static 'css/blog.css' %}" />
```
Comment (will-wow) [inline confero/templates/confero/base.html]: unused

File: docker-compose.yml [hunk 3]
```diff
@@ -13,7 +13,7 @@ services:
       - ./fec:/code/fec
       - ./htmlcov:/code/htmlcov
     environment:
-      DB_HOST: db
+      DB_HOSTNAME: db
```
Comment (will-wow) [inline docker-compose.yml]: this was just wrong, not sure how long that's been there

File: sandbox/bin/dump.sh [hunk 4]
```diff
@@ -1,4 +1,4 @@
 #! /bin/sh
 
 # TODO: This only works on Linux
-sudo -u postgres pg_dump -d confero -T fec_contribution > ./data/confero.sql  
+sudo -u postgres pg_dump -Fc -d confero -T fec_contribution > ./data/confero_archive
```
Comment (will-wow) [inline sandbox/bin/dump.sh]: dump a smaller "custom format" dumpfile

File: Dockerfile [hunk 5]
```diff
@@ -1,4 +1,4 @@
-FROM python:3.7-alpine AS base
+FROM python:3.7-slim-buster AS base
```
[the remark this answers is not in the document]
Comment (cgarretty) [reply Dockerfile]: This is solid. Slim-buster is my go to now.

File: .circleci/config.yml [hunk 6]
```diff
@@ -22,7 +22,7 @@ jobs:
       - run:
           command: |
             sudo pip install pipenv
-            pipenv install --dev
+            pipenv install --dev --system
```
[the remark this answers is not in the document]
Comment (cgarretty) [reply .circleci/config.yml]: smart.

File: .circleci/config.yml [hunk 7]
```diff
@@ -31,17 +31,17 @@ jobs:
...
           name: Initialize Elastic Beanstalk
-          command: pipenv run eb init -p docker project-confero
+          command: eb init -p docker project-confero
       - deploy:
           name: Deploy to Elastic Beanstalk Environment
           command: |
             if [ "${CIRCLE_BRANCH}" == "master" ]; then
-              pipenv run eb deploy --region us-east-2
+              eb deploy --region us-east-2
             fi
```
Comment (cgarretty) [inline .circleci/config.yml]: Just a thought, but shouldn't we have the `./manage.py collectstatic` command somewhere in here so stuff doesn't get left out by mistake?

File: docker-compose.yml (hunk 3 above)
Comment (cgarretty) [reply docker-compose.yml]: oops!

File: .circleci/config.yml (hunk 7 above)
Comment (cgarretty) [reply .circleci/config.yml]: aaaah, I see it in the Dockerfile now. Could that create weird issues since we're copying in the code, not mounting it as a volume? If it works with `./bin/docker-tests` it'll probably be fine, but I haven't seen it done like that.

Comment (will-wow) [reply .circleci/config.yml]: Yeah I don't love it either. But I spent some time trying to get this running in an `.ebextension` `container_command`, without success. Then I realized that all I was trying to do was to run this code in the docker container, so why not do it directly?

And we could do it in Circle, but looks like by default ` eb deploy` just deploys a zip of all your git committed files. There's a way to override that with a `.ebignore` file, but I could imagine that going wrong when we update the `gitignore` and forget to update the `.ebignore`. 

Also, I just noticed that `.ebextensions/db-migrate.config` seems to already be configured to run `collectstatic`, but it's apparently not working, since without this last line the static files don't get served.

Anyway tl;dr - maybe the next time we're messing with the EB config we can figure out a better way to do this
